gen/lib/gen/generator.rb

```diff
@@ -0,0 +1,113 @@
+require 'gen'
+require 'fileutils'
+require 'pathname'
+
+module Gen
+  class Generator
+    def self.run(project_name)
+      new(project_name).run
+    end
+
+    TEMPLATE_ROOT = File.expand_path('gen/template', Gen::ROOT)
+
+    VALID_PROJECT_NAME = /\A[a-z][a-z0-9]*\z/
+    private_constant :VALID_PROJECT_NAME
+
+    VENDOR_TRANSLATIONS = {
+      'Gemfile'            => false,
+      'exe/__app__-gems'   => false,
+      'exe/__app__-vendor' => 'exe/__app__',
+    }.freeze
+    private_constant :VENDOR_TRANSLATIONS
+
+    BUNDLER_TRANSLATIONS = {
+      'bin'                => false,
+      'bin/update-deps'    => false,
+      'exe/__app__-gems'   => 'exe/__app__',
+      'exe/__app__-vendor' => false,
+    }.freeze
+    private_constant :BUNDLER_TRANSLATIONS
+
+    def initialize(project_name)
+      raise(
+        CLI::Kit::Abort,
+        "project name must match {{bold:#{VALID_PROJECT_NAME}}} (but can be changed later)"
+      ) unless project_name =~ VALID_PROJECT_NAME
+      @project_name = project_name
+      @title_case = @project_name[0].upcase + @project_name[1..-1]
```

Comment:
Wouldn't this technically be `@project_name.split('/_-/).map { |part| part[0].upcase + part[1..-1] }.join`

Reply:
I wrote it to reject _ and - becuase I didn't want to have to deal with constant nesting for the generator in v1.

Reply:
`@project_name.sub(/^./, &:upcase)` seems more readable, despite the regex.